core/domain/suggestion_services_test.py

```diff
@@ -546,47 +594,51 @@ def test_accept_suggestion_raises_exception_if_suggestion_already_rejected(
             'The suggestion with id %s has already been accepted/rejected.' % (
                 self.suggestion_id)
         )
-        with self.assertRaisesRegex(Exception, expected_exception_regexp):
+        with self.assertRaisesRegex(Exception, expected_exception_regexp):  # type: ignore[no-untyped-call]
             suggestion_services.accept_suggestion(
-                self.suggestion_id, self.reviewer_id, self.COMMIT_MESSAGE, None)
+                self.suggestion_id, self.reviewer_id, self.COMMIT_MESSAGE, '')
 
         # Assert that the suggestion is still rejected.
         self.assert_suggestion_status(
             self.suggestion_id, suggestion_models.STATUS_REJECTED)
 
-    def test_accept_suggestion_invalid_suggestion_failure(self):
+    def test_accept_suggestion_invalid_suggestion_failure(self) -> None:
         self.mock_create_suggestion(self.target_id)
         suggestion = suggestion_services.get_suggestion_by_id(
             self.suggestion_id)
 
-        with self.assertRaisesRegex(
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             utils.ValidationError, 'Expected score_category to be of the form '
                                    'score_type.score_sub_type, received '
                                    'invalid_score_category'):
             with self.swap(
                 suggestion_registry.SuggestionEditStateContent,
                 'pre_accept_validate',
-                self.edit_before_pre_accept_validate(suggestion)):
+                # The `edit_before_pre_accept_validate` method does not return
+                # anything but we are using this method inside `with` clause,
+                # so MyPy expecting a return value and throwing an error for
+                # the same. Thus to avoid the error, we used ignore here.
+                self.edit_before_pre_accept_validate(suggestion)):  # type: ignore[func-returns-value]
```

Comment:
Why is MyPy expecting return value in a `swap`?

Reply:
Ah, actually this is not related to `swap` particular, because:
Lets consider the following cases:
```python
def foo(x: int) -> None:
    x = 5

var_1 = foo(39)        # Here MyPy throws an error because Mypy assumes that the return value of methods that return None should not be used.


def foo_one(x: str) -> str
    return x

foo_one(foo(35))       # Here also MyPy throws an error, because Mypy assumes that the return value of methods that return None should not be used.
```
So, a similar is happening with the swap.
`self.edit_before_pre_accept_validate(suggestion)` is used inside swap so Mypy assumes that it will return something.

Reply:
Reference: https://github.com/python/mypy/issues/6549
It is not an exact issue but I hope it makes things clearer.

Reply:
Hmm, okay I think I understand this error, but I'm still confused why it is used like this here, if we want to swap `suggestion_registry.SuggestionEditStateContent.pre_accept_validate` with `None` we don't need to use `self.edit_before_pre_accept_validate(suggestion)`. Should just `self.edit_before_pre_accept_validate` be used? Or maybe we call `self.edit_before_pre_accept_validate(suggestion)` outside the `swap`.

Reply:
Actually, we are not replacing it for the None value, instead, we are changing the implementation of `SuggestionEditStateContent.pre_accept_validate` with `self.edit_before_pre_accept_validate(suggestion)` for this testing purpose.

I also tried to refactor it but didn't find any working solution.

Reply:
The problem is that when we swap elsewhere we swap with the function and not the function call. `swap(a, 'b', c)` not `swap(a, 'b', c())` that is why is this so confusing.

Reply:
We cannot do
```python
            with self.swap(
                suggestion_registry.SuggestionEditStateContent,
                'pre_accept_validate',
                self.edit_before_pre_accept_validate):
```
Because  edit_before_pre_accept_validate is implemented as below:
```python
    def edit_before_pre_accept_validate(
        self, suggestion: suggestion_registry.BaseSuggestion
    ) -> None:
        """Edits suggestion immediately before pre-accept validation."""
        suggestion.score_category = 'invalid_score_category'
        suggestion.pre_accept_validate()
```
so if we replace `edit_before_pre_accept_validate` with `pre_accept_validate`, then whenever we call `pre_accept_validate` a `edit_before_pre_accept_validate` will be called which in turn will call `pre_accept_validate` and the cycle goes on.

But if we call `self.edit_before_pre_accept_validate` then everything works fine, so I moved `self.edit_before_pre_accept_validate` outside the swap.